ui/src/app/machines/views/MachineDetails/MachineTests/MachineTestsTable/MachineTestsTable.tsx

```diff
@@ -16,19 +20,183 @@ const isSuppressible = (result: ScriptResult) =>
+                    type="checkbox"
+                    id={`suppress-${result.id}`}
+                    data-test="suppress-script-results"
+                    label=" "
+                    checked={result.suppressed}
+                    onChange={() => {
+                      result.suppressed
+                        ? dispatch(
+                            machineActions.unsuppressScriptResults(machineId, [
+                              result,
+                            ])
+                          )
+                        : dispatch(
+                            machineActions.suppressScriptResults(machineId, [
+                              result,
+                            ])
+                          );
+                    }}
+                  />
+                </>
+              ) : null}
+            </>
+          ),
+        },
+        {
+          content: (
+            <span data-test="name">
+              <Icon
+                name={
+                  result.status === scriptStatus.PASSED
+                    ? ICONS.success
+                    : ICONS.error
+                }
+              />
+              {result.name || "—"}
+            </span>
+          ),
+        },
+        {
+          content: <span data-test="tags">{result.tags}</span>,
+        },
+        {
+          content: <span data-test="runtime">{result.runtime}</span>,
+        },
+        {
+          content: <span data-test="date">{result.updated}</span>,
+        },
+        {
+          content: <span data-test="status">{result.status_name}</span>,
+        },
+        {
+          content: renderActions(
+            result,
+            showPreviousTests,
+            hidePreviousTests,
+            hasHistory,
+            isExpanded
+          ),
+          className: "u-align--right",
+        },
+      ],
+      expandedContent: (
+        <table role="grid" className="p-main-table">
+          <tbody>
+            {history[result.id]?.map((item) => {
+              if (item.id !== result.id) {
+                return (
+                  <tr
+                    role="row"
+                    data-test="script-result-history"
+                    key={item.id}
+                  >
+                    <td role="gridcell"></td>
+                    <td role="gridcell"></td>
+                    <td role="gridcell"></td>
+                    <td role="gridcell">{item.runtime}</td>
+                    <td role="gridcell">{item.updated}</td>
+                    <td role="gridcell">{item.status_name}</td>
+                    <td role="gridcell"></td>
+                  </tr>
+                );
+              } else {
+                return null;
+              }
+            })}
+          </tbody>
+        </table>
+      ),
+      key: result.id,
+      sortData: {
+        name: result.name,
+        date: result.updated,
+      },
+    });
+  });
+
   return (
     <>
       <MainTable
+        expanding
```

Comment:
We've also been using the className "p-table-expanding--light" for expanding tables, which makes the table row and expanded cell white for the expanded rows.